scripts/xtensa-build-zephyr.py

```diff
@@ -902,6 +908,69 @@ def build_platforms():
 			symlinks=True, ignore_dangling_symlinks=True, dirs_exist_ok=True)
 
 
+def install_lib(platform, sof_output_dir, abs_build_dir):
+	"""[summary] Sign loadable llext modules, if any, copy them to the
+	deployment tree and create UUID links for the kernel to find and load
+	them."""
+
+	global signing_key
+	global platform_wconfig
+
+	with os.scandir(str(abs_build_dir)) as iter:
+		rimage_executable = platform_wconfig.get("rimage.path")
+		if rimage_executable is None:
+			rimage_executable = platform_wconfig.get("rimage.path", None,
+								 west_config.ConfigFile.LOCAL)
+
+		for entry in iter:
+			if (not entry.is_dir or
+			    not entry.name.endswith('_llext')):
+				continue
+			uuids = pathlib.Path(entry.path) / 'llext.uuid'
+			if not os.path.exists(uuids):
+				continue
+
+			# replace '_llext' with '.llext', e.g.
+			# eq_iir_llext/eq_iir.llext
+			llext_base = entry.name[:-6]
+			llext_file = llext_base + '.llext'
+
+			sof_lib_dir = sof_output_dir / '..' / 'sof-ipc4-lib' / platform
+			if args.key_type_subdir != "none":
+				sof_lib_dir = sof_lib_dir / args.key_type_subdir
+			sof_lib_dir.mkdir(parents=True, exist_ok=True)
+			dst = sof_lib_dir / llext_file
+
+			rimage_cfg = pathlib.Path(entry.path) / 'rimage_config.toml'
+			llext_input = pathlib.Path(entry.path) / str(llext_base + '.so')
+			llext_output = pathlib.Path(entry.path) / llext_file
+
+			sign_cmd = [str(rimage_executable), "-o", str(llext_output),
+				    "-e", "-c", str(rimage_cfg),
+				    "-k", str(signing_key), "-l", "-r",
+				    str(llext_input)]
+			execute_command(sign_cmd, cwd=west_top)
+
+			# Explicitly rely on default exception handling
+			fdst = open(str(dst), 'wb')
+			fllext = open(str(llext_output), 'rb')
+			fman = open(str(llext_output) + '.xman', 'rb')
+
+			shutil.copyfileobj(fman, fdst)
+			shutil.copyfileobj(fllext, fdst)
+
+			fdst.close()
+			fman.close()
+			fllext.close()
+
+			with open(uuids, 'r') as f_llext:
+				lines = f_llext.readlines()
+				for line in lines:
+					symlink_or_copy(sof_lib_dir, llext_file,
+							sof_lib_dir, line.strip() + '.bin')
+				f_llext.close()
```

Comment:
Would be nice to have example output in the commit message to explain what the intention is.

Reply:
@kv2019i sure, here's the output when printing a deployment tree during the build:
```
build-sof-staging  
|-- sof  
|   +-- intel  
|       |-- sof-ipc4
...
|       +-- sof-ipc4-lib  
|           +-- mtl  
|               +-- dbgkey  
|                   |-- 167A961E-8AE4-11EA-89F1-000C29CE1635.bin  -> smart_amp_test.llext 
|                   |-- 39656EB2-3B71-4049-8D3F-F92CD5C43C09.bin  -> mixin_mixout.llext 
|                   |-- 3C56505A-24D7-418F-BDDC-C1F5A3AC2AE0.bin  -> mixin_mixout.llext 
|                   |-- 5150C0E6-27F9-4EC8-8351-C705B642D12F.bin  -> eq_iir.llext 
|                   |-- eq_iir.llext  
|                   |-- mixin_mixout.llext  
|                   +-- smart_amp_test.llext
```
The bottom 3 lines above are llext modules and the four lines above them are UUID links for driver loading

Reply:
> example output in the commit message

Maybe what would be even better is to switch `.github/workflows/llext.yml` to invoke all this new code instead of `west build`? This would not just show sample output on Github, but also provide test coverage immediately.

Reply:
> switch `.github/workflows/llext.yml` to invoke all this new code

@marc-hb propose as a follow-up

Reply:
> > switch `.github/workflows/llext.yml` to invoke all this new code
> 
> @marc-hb propose as a follow-up

https://github.com/thesofproject/sof/pull/9013/commits/d6b568cbcb815d2708a4b8cb031999f913a027ec